pkg/kubelet/kubelet.go

```diff
@@ -521,6 +521,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		containerManager:     kubeDeps.ContainerManager,
 		containerRuntimeName: containerRuntime,
 		nodeIP:               parsedNodeIP,
+		nodeIPValidator:      validateNodeIP,
```

Comment:
I think this is a pretty clean way of allowing validateNodeIP() to be mocked for testing.  The alternative I can think of is to create a kubelet wrapper for `net.InterfaceAddrs()` to allow you test get testing coverage for the `validateNodeIP()` function itself.